superbench/benchmarks/docker_benchmarks/docker_base.py

```diff
@@ -36,24 +44,87 @@ def _preprocess(self):
         Return:
             True if _preprocess() succeed.
         """
-        return super()._preprocess()
+        if not super()._preprocess():
+            return False
+
+        if self._image is None:
+            self._result.set_return_code(ReturnCode.DOCKERBENCHMARK_IMAGE_NOT_SET)
+            logger.error('The image url is not set - benchmark: {}.'.format(self._name))
+            return False
+
+        if self._container is None:
+            self._result.set_return_code(ReturnCode.DOCKERBENCHMARK_CONTAINER_NOT_SET)
+            logger.error('The container name is not set - benchmark: {}.'.format(self._name))
+            return False
+
+        output = run_command('docker pull {}'.format(self._image))
+        if output.returncode != 0:
+            self._result.set_return_code(ReturnCode.DOCKERBENCHMARK_IMAGE_PULL_FAILURE)
+            logger.error(
+                'Docker benchmark pull image failed - benchmark: {}, error message: {}.'.format(
+                    self._name, output.stdout
+                )
+            )
+            return False
+
+        return True
+
+    def _postprocess(self):
+        """Postprocess/cleanup operations after the benchmarking.
+
+        Return:
+            True if _postprocess() succeed.
+        """
+        rm_containers = 'docker rm $(docker stop {}))'.format(self._container)
```

Comment:
Because we don't have `docker run` in the docker_base, but we have `docker rm` and `docker stop`. Is it a little bit strange?

Reply:
`docker run` need some customized arguments and is set into `self._commands` of derived class, for most of benchmarks, we only need one `docker run` command to launch the benchmark, take ONNX's benchmark for example:
```
docker run -it --rm \
  --privileged \
  --ipc=host \
  --device=/dev/kfd \
  --device=/dev/dri \
  --group-add video \
  --cap-add=SYS_PTRACE \
  --security-opt seccomp=unconfined \
  -v /myhost/working/path:/workspace/ \
  -v /myhost/working/path/logs:/logs/ \
  --workdir /workspace/ \
  sukha.azurecr.io/onnxruntime/rocm4.2:benchmark
```